coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -0,0 +1,150 @@
+package filodb.coordinator.queryplanner
+
+import com.typesafe.config.ConfigFactory
+
+import filodb.coordinator.queryplanner.LogicalPlanUtils._
+import filodb.core.metadata.Dataset
+import filodb.core.query.{PromQlQueryParams, QueryContext}
+import filodb.query.{BinaryJoin, LabelValues, LogicalPlan, PeriodicSeriesPlan, SeriesKeysByFilters}
+import filodb.query.exec._
+
+case class PartitionAssignment(partitionName: String, endPoint: String, timeRange: TimeRange)
+
+trait PartitionLocationProvider {
+
+  def getPartitions(routingKey: Map[String, String], timeRange: TimeRange): List[PartitionAssignment]
+  def getAuthorizedPartitions(timeRange: TimeRange): List[PartitionAssignment]
+}
+
+class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider,
+                            localPlanner: QueryPlanner,
```

Comment:
Rename param to localPartitionPlanner. Consider making it a SinglePartitionPlanner for type safety.